server.js

```diff
@@ -7,22 +7,72 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number,
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
+// Start defining your routes here -> code along with Jennie
+app.get('/', (_, res) => {
   res.send('Hello world')
 })
 
+app.get('/', async (_, res) => {
+  const thoughts = await Thought.find()
+    .sort({ createdAt: 'desc' })
+    .limit(20)
+  res.json(thoughts)
+})
+
+app.post('/', async (req, res) => {
+  const thought = new Thought({
+    message: req.body.message,
+    hearts: 0
```

Comment:
you don't need to specify hearts: 0 here since your getting that as a default from the mongoose model